crypto/store/loader_file.c

```diff
@@ -764,20 +764,22 @@ static OSSL_STORE_LOADER_CTX *file_open(const OSSL_STORE_LOADER *loader,
         }
     } else {
         BIO *buff = NULL;
-        char peekbuf[4096];
+        char *peekbuf = OPENSSL_malloc(4096);
```

Comment:
Maybe it's my background of kernel work coming out, but 4k feels like a lot to put on the stack, yes.
Looking more closely this is in file_open, so we wouldn't recurse directly here when handling OSSL_STORE_INFO_EMBEDDED objects, but there may well be other things in the call stack that take up lots of space.  Some threading implementations give pretty small stacks by default, IIRC.